packages/collaborative-drive/src/tokens.ts

```diff
@@ -21,18 +22,7 @@ export const IGlobalAwareness = new Token<IAwareness>(
   '@jupyter/collaboration:IGlobalAwareness'
 );
 
-/**
- * A document factory for registering shared models
- */
-export type SharedDocumentFactory = (
-  options: Contents.ISharedFactoryOptions
-) => YDocument<DocumentChange>;
-
-/**
- * A Collaborative implementation for an `IDrive`, talking to the
- * server using the Jupyter REST API and a WebSocket connection.
- */
-export interface ICollaborativeDrive extends Contents.IDrive {
```

Comment:
Looks like we'll want to log that somewhere in the changelog as API breaking change, since other extensions may currently depend on `ICollaborativeDrive`?

For instance it's the case of JupyterCAD and JupyterGIS:
- https://github.com/jupytercad/JupyterCAD/blob/c246bf28480b8df1cbe2d20a0bbcec40aa881a5e/packages/base/src/panelview/rightpanel.tsx#L15
- https://github.com/geojupyter/jupytergis/blob/3bfd970aa7a513ebad30d31acf24adef11b6cf32/python/jupytergis_core/src/factory.ts#L2

Reply:
It looks like the transition will be smooth because:
- both extensions used `ICollaborativeDrive` as an optional dependency (after all it was just added)
- neither of the extensions actually relied on the `Drive` part of `ICollaborativeDrive` - the same properties that they needed will be available on `ICollaborativeContentProvider`

So when cutting the release we will just need to add a note that migration from `ICollaborativeDrive` to `ICollaborativeContentProvider` token for access to shared factory and forks will be required. I do not see a good way to do that right now as the changelog for next version is not there yet - is it ok if we add it during/just after cutting the release?